certification/engine/operatorsdk.go

```diff
@@ -0,0 +1,184 @@
+package engine
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification/errors"
+	fileutils "github.com/redhat-openshift-ecosystem/openshift-preflight/certification/internal/utils/file"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/cli"
+	log "github.com/sirupsen/logrus"
+)
+
+func NewOperatorSdkEngine() *cli.OperatorSdkEngine {
+	var engine cli.OperatorSdkEngine = operatorSdkEngine{}
+	return &engine
+}
+
+type operatorSdkEngine struct{}
+
+func (o operatorSdkEngine) Scorecard(image string, opts cli.OperatorSdkScorecardOptions) (*cli.OperatorSdkScorecardReport, error) {
+	cmdArgs := []string{"scorecard"}
+	if opts.OutputFormat == "" {
+		opts.OutputFormat = "json"
+	}
+	cmdArgs = append(cmdArgs, "--output", opts.OutputFormat)
+	if opts.Selector != nil {
+		for _, selector := range opts.Selector {
+			cmdArgs = append(cmdArgs, fmt.Sprintf("--selector=%s", selector))
+		}
+	}
+	if opts.Kubeconfig != "" {
+		cmdArgs = append(cmdArgs, "--kubeconfig", opts.Kubeconfig)
+	}
+	if opts.Namespace != "" {
+		cmdArgs = append(cmdArgs, "--namespace", opts.Namespace)
+	}
+	if opts.ServiceAccount != "" {
+		cmdArgs = append(cmdArgs, "--service-account", opts.ServiceAccount)
+	}
+
+	configFile, err := createScorecardConfigFile()
+	if err != nil {
+		log.Error("could not create scorecard config file", err)
+		return nil, err
+	}
+	defer os.Remove(configFile)
```

Comment:
I think the defer statement should be placed right after line 46. Otherwise if an error occurred while executing `createScorecardConfigFile`, the Scorecard method returns and the execution will never reach this line to push the deferred method to the stack.